src/main/java/com/mongodb/kafka/connect/source/schema/BsonDocumentToSchema.java

```diff
@@ -36,6 +36,19 @@
   private static final Schema DEFAULT_INFER_SCHEMA_TYPE = Schema.OPTIONAL_STRING_SCHEMA;
   public static final String SCHEMA_NAME_TEMPLATE = "inferred_name_%s";
 
+  public static Schema inferDocumentSchema(final BsonDocument document) {
+    SchemaBuilder builder = SchemaBuilder.struct();
+    if (document.containsKey(ID_FIELD)) {
```

Comment:
Why does _id have to go first?

Reply:
It doesn't have to but we follow the idom that the _id field is always the first field in the documents.